### PR TITLE
Upgrade kernel to 5.15.136.1 and add AMDGPU support

### PR DESCRIPTION
**Pull request by rlmenge**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Upgrade kernel to 5.15.136.1 and add AMDGPU support. This PR also adds cpio as a BuildRequires and cleans up the changelog for specialized kernel packages

This upgrade includes an additional feature branch for the AMDGPU drivers. This has led to additional config changes to enable.

You can see this change by looking at the [commit ](https://github.com/microsoft/CBL-Mariner-Linux-Kernel/commit/d9d3c62870abf05849104bfce24935be520b1e19)for the kernel source or the [merge file](https://github.com/microsoft/CBL-Mariner-Linux-Kernel/blob/rolling-lts/mariner-2/5.15.136.1/MSFT-Merge/log) for the source here 

The orignal autopatcher [PR](https://github.com/microsoft/CBL-Mariner/pull/6593) was failing with build errors and differing configs. I wanted to preserve this data for historical and debugging purposes but I also had to rebase. Therefore I made a new PR.

###### Change Log  

- Upgrade to 5.15.136.1

###### Does this affect the toolchain?  

**YES**

###### Test Methodology

- Pipeline buddybuild : https://dev.azure.com/mariner-org/mariner/_build/results?buildId=443133&view=results

- vhdx boots in a hyperv vm

### CODE REVIEW

**Review by neha170 — COMMENTED** [review]
Just wondering why the auto upgrade pipeline was not able to handle this, thanks

**Comment by rlmenge** [comment]
> Just wondering why the auto upgrade pipeline was not able to handle this, thanks

The orignal autopatcher [PR](https://github.com/microsoft/CBL-Mariner/pull/6593) was failing with build errors and differing configs. I wanted to preserve this data for historical and debugging purposes but I also had to rebase. Therefore I made a new PR :) . Also updated the description with this

**Review by henryli001 — COMMENTED** [review]
Just would like to clarify what "AMDGPU support" refers to here? 

I looked at the related kernel configs and it seems that the configs that are responsible for building the drivers such as DRM_AMDGPU, DRM_AMDGPU_SI and DRM_AMDGPU_CIK have already been enabled. So does it mean that the kernel is already capable of building the drivers and this PR adds additional support for AMD GPU like enabling display engine config? (Please correct me if I'm wrong here).

**Comment by rlmenge** [comment]
> Just would like to clarify what "AMDGPU support" refers to here?
> 
> I looked at the related kernel configs and it seems that the configs that are responsible for building the drivers such as DRM_AMDGPU, DRM_AMDGPU_SI and DRM_AMDGPU_CIK have already been enabled. So does it mean that the kernel is already capable of building the drivers and this PR adds additional support for AMD GPU like enabling display engine config? (Please correct me if I'm wrong here).

It is a completely new feature branch merged in. The code is not fully upstream yet which is why I called it out. The config changes are a reflection of the new code being added to the kernel :) 

You can see this change by looking at the [commit ](https://github.com/microsoft/CBL-Mariner-Linux-Kernel/commit/d9d3c62870abf05849104bfce24935be520b1e19)for the kernel source or the [merge file](https://github.com/microsoft/CBL-Mariner-Linux-Kernel/blob/rolling-lts/mariner-2/5.15.136.1/MSFT-Merge/log) for the source here 

I have also updated the description with this

**Comment by rlmenge** [comment]
kernel-hci and kernel-azure are not building as expected. Need to analyze their configs before this merges

File: SPECS-SIGNED/kernel-azure-signed/kernel-azure-signed.spec [hunk 1]
```diff
@@ -153,13 +153,16 @@ ln -sf linux-%{uname_r}.cfg /boot/mariner.cfg
...
+* Thu Oct 26 2023 Rachel Menge <rachelmenge@microsoft.com> - 5.15.136.1-1
+- Upgrade to 5.15.136.1
+
 * Tue Oct 17 2023 CBL-Mariner Servicing Account <cblmargh@microsoft.com> - 5.15.135.1-1
 - Auto-upgrade to 5.15.135.1
 
 * Tue Sep 26 2023 CBL-Mariner Servicing Account <cblmargh@microsoft.com> - 5.15.133.1-1
 - Auto-upgrade to 5.15.133.1
 
-* Tue Sep 22 2023 Cameron Baird <cameronbaird@microsoft.com> - 5.15.131.1-3
```
Comment (Camelron) [inline SPECS-SIGNED/kernel-azure-signed/kernel-azure-signed.spec:162]: sorry about that...

**Review by Camelron — APPROVED** [review]
lgtm

**Review by christopherco — CHANGES_REQUESTED** [review]

File: SPECS/kernel/kernel.spec [hunk 2]
```diff
@@ -44,6 +44,7 @@ BuildRequires:  audit-devel
 BuildRequires:  bash
 BuildRequires:  bc
 BuildRequires:  build-essential
+BuildRequires:  cpio
```
[the remark this answers is not in the document]
Comment (rlmenge) [reply SPECS/kernel/kernel.spec:47]: done :)

File: SPECS/kernel-hci/kernel-hci.spec [hunk 3]
```diff
@@ -17,7 +17,7 @@
 %define config_source %{SOURCE1}
 Summary:        Linux Kernel for HCI
 Name:           kernel-hci
-Version:        5.15.135.1
+Version:        5.15.136.1
```
Comment (rlmenge) [inline SPECS/kernel-hci/kernel-hci.spec:20]: Wait on merging until kernel-hci validation from @vinceaperri has passed

**Comment by rlmenge** [comment]
Looking to add AMDGPU drivers in a seperate kernel. Closing this PR

**Pull request closed without merging**